src/main/java/htsjdk/beta/io/bundle/Bundle.java

```diff
@@ -38,7 +38,9 @@
 public class Bundle implements Iterable<BundleResource>, Serializable {
     private static final long serialVersionUID = 1L;
 
-    private final Map<String, BundleResource> resources = new LinkedHashMap<>();
+    // don't use  LinkedHashMap here; using HashMap resolves unnatural resource ordering issues that arise
```

Comment:
Hm. I was having problems when roundtripping these through JSON, but I can no longer reproduce the issue, so reverting to our beloved LinkedHashMap.

Reply:
Long Live LinkedHashMap!

Reply:
So I realized what the issue was here. `JSONObject` internally uses a `HashMap` (implying, I think, that JSON doesn't preserve the serialized order of JSON attributes), so when you roundtrip through JSON, the iteration order from JSON is based on the `HashMap` order. If we use `LinkedHashMap` in `Bundle`, then the order after a roundtrip gets scrambled, and tests fail (but only for some cases because sometimes the roundtrip order matches and sometimes it differs).

Reply:
Ahah, that makes sense.  Seems like a weird decision to not maintain internal order, but it's good to know about.  Cana we change our tests to use an order independent comparison?

Reply:
I added an resource-order-independent equals method for use in the tests.